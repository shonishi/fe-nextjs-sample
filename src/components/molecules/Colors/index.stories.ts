import type { Meta, StoryObj } from '@storybook/react';
import Colors from '.';

const meta: Meta<typeof Colors> = {
  component: Colors,
};

export default meta;
type Story = StoryObj<typeof Colors>;

export const MultipleElements: Story = {
  args: {
    colors: [
      {
        id: 1,
        name: 'Pale yellow',
        class: 'bg-yellow-50',
        selectedClass: 'ring-gray-400',
      },
      {
        id: 2,
        name: 'Yellow',
        class: 'bg-yellow-100',
        selectedClass: 'ring-gray-400',
      },
      {
        id: 3,
        name: 'Dark yellow',
        class: 'bg-yellow-300',
        selectedClass: 'ring-gray-900',
      },
    ],
    selectedColorId: 1,
    changeColor: undefined,
  },
};

export const SingleElements: Story = {
  args: {
    colors: [
      {
        id: 1,
        name: 'Pale yellow',
        class: 'bg-yellow-50',
        selectedClass: 'ring-gray-400',
      },
    ],
    selectedColorId: 1,
    changeColor: undefined,
  },
};

export const EmptyElements: Story = {
  args: {
    colors: [],
    selectedColorId: 0,
    changeColor: undefined,
  },
};
