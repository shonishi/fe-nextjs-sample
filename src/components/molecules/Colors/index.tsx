import { RadioGroup } from '@headlessui/react';

export interface Props {
  colors: {
    id: number;
    name: string;
    class: string;
    selectedClass: string;
  }[];
  selectedColorId: number;
  changeColor: (e: { id: number }) => {
    payload: number;
    type: string;
  };
}

export default function Colors({
  colors,
  selectedColorId,
  changeColor,
}: Props) {
  const selectedColor = colors.find((color) => color.id === selectedColorId);
  return (
    <div>
      <h3 className="text-sm font-medium text-gray-900">Color</h3>
      <RadioGroup value={selectedColor} onChange={changeColor} className="mt-4">
        <RadioGroup.Label className="sr-only">Choose a color</RadioGroup.Label>
        <div className="flex items-center space-x-3">
          {colors.map((color) => (
            <RadioGroup.Option
              key={color.name}
              value={color}
              className={({ active, checked }) =>
                [
                  color.selectedClass,
                  active && checked ? 'ring ring-offset-1' : '',
                  !active && checked ? 'ring-2' : '',
                  'relative -m-0.5 flex cursor-pointer items-center justify-center rounded-full p-0.5 focus:outline-none',
                ]
                  .filter(Boolean)
                  .join(' ')
              }
            >
              <RadioGroup.Label as="span" className="sr-only">
                {color.name}
              </RadioGroup.Label>
              <span
                aria-hidden="true"
                className={[
                  color.class,
                  'h-8 w-8 rounded-full border border-black border-opacity-10',
                ].join(' ')}
              />
            </RadioGroup.Option>
          ))}
        </div>
      </RadioGroup>
    </div>
  );
}
