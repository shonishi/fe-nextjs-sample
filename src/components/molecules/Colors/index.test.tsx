import Colors from '.';
import renderer from 'react-test-renderer';

describe('Colors', () => {
  it('colorsの要素が複数の場合、全て表示されること', async () => {
    const tree = renderer
      .create(
        <Colors
          {...{
            colors: [
              {
                id: 1,
                name: 'Pale yellow',
                class: 'bg-yellow-50',
                selectedClass: 'ring-gray-400',
              },
              {
                id: 2,
                name: 'Yellow',
                class: 'bg-yellow-100',
                selectedClass: 'ring-gray-400',
              },
              {
                id: 3,
                name: 'Dark yellow',
                class: 'bg-yellow-300',
                selectedClass: 'ring-gray-900',
              },
            ],
            selectedColorId: 1,
            changeColor: jest.fn(),
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('colorsの要素が1つの場合、1つだけ表示されること', async () => {
    const tree = renderer
      .create(
        <Colors
          {...{
            colors: [
              {
                id: 1,
                name: 'Pale yellow',
                class: 'bg-yellow-50',
                selectedClass: 'ring-gray-400',
              },
            ],
            selectedColorId: 1,
            changeColor: jest.fn(),
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('colorsが空の場合、何も表示されないこと', async () => {
    const tree = renderer
      .create(
        <Colors
          {...{
            colors: [],
            selectedColorId: 0,
            changeColor: jest.fn(),
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
