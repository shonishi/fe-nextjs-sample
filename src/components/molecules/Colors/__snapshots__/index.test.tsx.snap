// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors colorsが空の場合、何も表示されないこと 1`] = `null`;

exports[`Colors colorsの要素が1つの場合、1つだけ表示されること 1`] = `
<div>
  <h3
    className="text-sm font-medium text-gray-900"
  >
    Color
  </h3>
  <div
    aria-labelledby="headlessui-label-:r9:"
    className="mt-4"
    id="headlessui-radiogroup-:r8:"
    onKeyDown={[Function]}
    role="radiogroup"
  >
    <label
      className="sr-only"
      id="headlessui-label-:r9:"
    >
      Choose a color
    </label>
    <div
      className="flex items-center space-x-3"
    >
      <div
        aria-checked="true"
        aria-labelledby="headlessui-label-:rb:"
        className="ring-gray-400 ring-2 relative -m-0.5 flex cursor-pointer items-center justify-center rounded-full p-0.5 focus:outline-none"
        data-headlessui-state="checked"
        id="headlessui-radiogroup-option-:ra:"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        role="radio"
        tabIndex={0}
      >
        <span
          className="sr-only"
          id="headlessui-label-:rb:"
        >
          Pale yellow
        </span>
        <span
          aria-hidden="true"
          className="bg-yellow-50 h-8 w-8 rounded-full border border-black border-opacity-10"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Colors colorsの要素が複数の場合、全て表示されること 1`] = `
<div>
  <h3
    className="text-sm font-medium text-gray-900"
  >
    Color
  </h3>
  <div
    aria-labelledby="headlessui-label-:r1:"
    className="mt-4"
    id="headlessui-radiogroup-:r0:"
    onKeyDown={[Function]}
    role="radiogroup"
  >
    <label
      className="sr-only"
      id="headlessui-label-:r1:"
    >
      Choose a color
    </label>
    <div
      className="flex items-center space-x-3"
    >
      <div
        aria-checked="true"
        aria-labelledby="headlessui-label-:r3:"
        className="ring-gray-400 ring-2 relative -m-0.5 flex cursor-pointer items-center justify-center rounded-full p-0.5 focus:outline-none"
        data-headlessui-state="checked"
        id="headlessui-radiogroup-option-:r2:"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        role="radio"
        tabIndex={0}
      >
        <span
          className="sr-only"
          id="headlessui-label-:r3:"
        >
          Pale yellow
        </span>
        <span
          aria-hidden="true"
          className="bg-yellow-50 h-8 w-8 rounded-full border border-black border-opacity-10"
        />
      </div>
      <div
        aria-checked="false"
        aria-labelledby="headlessui-label-:r5:"
        className="ring-gray-400 relative -m-0.5 flex cursor-pointer items-center justify-center rounded-full p-0.5 focus:outline-none"
        data-headlessui-state=""
        id="headlessui-radiogroup-option-:r4:"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        role="radio"
        tabIndex={-1}
      >
        <span
          className="sr-only"
          id="headlessui-label-:r5:"
        >
          Yellow
        </span>
        <span
          aria-hidden="true"
          className="bg-yellow-100 h-8 w-8 rounded-full border border-black border-opacity-10"
        />
      </div>
      <div
        aria-checked="false"
        aria-labelledby="headlessui-label-:r7:"
        className="ring-gray-900 relative -m-0.5 flex cursor-pointer items-center justify-center rounded-full p-0.5 focus:outline-none"
        data-headlessui-state=""
        id="headlessui-radiogroup-option-:r6:"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        role="radio"
        tabIndex={-1}
      >
        <span
          className="sr-only"
          id="headlessui-label-:r7:"
        >
          Dark yellow
        </span>
        <span
          aria-hidden="true"
          className="bg-yellow-300 h-8 w-8 rounded-full border border-black border-opacity-10"
        />
      </div>
    </div>
  </div>
</div>
`;
